src/line_printer.cc

```diff
@@ -102,7 +103,22 @@ void LinePrinter::PrintOrBuffer(const char* data, size_t size) {
   } else {
     // Avoid printf and C strings, since the actual output might contain null
     // bytes like UTF-16 does (yuck).
+#ifdef _WIN32
+    size_t bytes_written = fwrite(data, 1, size, stdout);
+    if (bytes_written == 0 &&
```

Comment:
It would make the code faster, but since this is an edge case and the first bisection will likely be enough already, I don't think that it's worth it (I'm assuming the while version would be longer and harder to reason about).

How would it make the code more robust?